SCL polygon exports

thanrobinson opened the pull request:
- added exporting of unique polygon types
- added time stamp to asset id
- comments remain for key todos

Commit:
added functions for distance and area to pixels 

dynamic distance and area to pixels based on scale

thanrobinson committed:
scl_170

export each landscape type individually, adds time stamp to asset id

saanobhaai commented on src/task_scl_eff_pot_hab.py:
```diff
@@ -86,14 +100,25 @@ def __init__(self, *args, **kwargs):
         self.set_aoi_from_ee(self.inputs["probability"]["ee_path"])
 
     def calc(self):
+        def distance_km_to_pixels(distance_km, image_resolution):
```
I would move these methods into the class root and use class properties (like `self.scale`) as defaults, where possible.

saanobhaai commented on src/task_scl_eff_pot_hab.py:
```diff
@@ -55,20 +49,40 @@ class SCLPolygons(SCLTask):
+            "size": 3,
+            "probability": 1,
+            "effort": 1,
+        },  # large, high probabiliyt, has survey effor
```
spelling

saanobhaai commented on src/task_scl_eff_pot_hab.py:
```diff
@@ -115,35 +140,64 @@ def density_to_patch_size(ft):
+
+        connected_str_hab = str_hab_mask.connectedPixelCount(
+            str_hab_connected_pixels, str_hab_resolution
+        ).reproject("EPSG:4326", None, str_hab_resolution)
```
`self.crs` -- maybe do global search/replace

saanobhaai commented on src/task_scl_eff_pot_hab.py:
```diff
@@ -206,38 +261,76 @@ def dilate(image, distance):
-                reducer=ee.Reducer.max(),
-                geometry=self.extent,
+                reducer=ee.Reducer.max(),  # TODO: need to consider reducer for each band to delineate polygons
+                geometry=ee.Geometry.MultiPolygon(self.aoi).bounds(),
```
`ee.Geometry.MultiPolygon(self.aoi, proj=self.crs, geodesic=False)`
or, couldn't we just do 
`ee.Geometry.Polygon(self.extent)`
?

saanobhaai commented on src/task_scl_eff_pot_hab.py:
```diff
@@ -206,38 +261,76 @@ def dilate(image, distance):
+                ee.Filter.gte(
+                    "probability", self.scl_poly_filters["scl_survey"]["probability"]
+                ),
+                ee.Filter.lte("effort", self.scl_poly_filters["scl_survey"]["effort"]),
```
With the current filters structure, there is a logical possibility of overlaps (e.g. here we're using `lte` and above the effort threshold is using `gte`; one of these should probably leave out the `e`). Wondering if a simple filter structure would make sense?

saanobhaai reviewed (changes requested):
This is brilliant. Just a few things to maybe look at.

thanrobinson committed:
Changes based on PR Comments

spelling, moved key functions outside of calc, added polygon delineation thresholds, built filters in init
